src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.vocabulary.RDF;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.LPG2RDFConfiguration;
+import se.liu.ida.hefquin.vocabulary.LPG2RDF;
+
+public class LPG2RDFConfigurationReader {
+
+    public static LPG2RDFConfigurationReader instance = new LPG2RDFConfigurationReader();
+
+    protected LPG2RDFConfigurationReader(){}
+
+    public static LPG2RDFConfiguration readFromFile(final String filename ) {
+        return instance.parseLPG2RDFConf(filename);
+    }
+
+    public static LPG2RDFConfiguration readFromModel(final Model lpg2Rdf ) {
+        return instance.parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfiguration parseLPG2RDFConf( final String filename ) {
+        final Model lpg2Rdf = RDFDataMgr.loadModel(filename);
+        return parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfiguration parseLPG2RDFConf(final Model lpg2Rdf ) {
+
+        final ResIterator lpg2rdfConfigs = lpg2Rdf.listResourcesWithProperty(RDF.type, LPG2RDF.LPGtoRDFConfiguration);
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+        if(lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("More than one instance of LPGtoRDFConfiguration!");
+        }
+
+        final String validDataType = lpg2Rdf.getNsPrefixURI("xsd").concat("anyURI");
+
+        final Node label = getLabelPredicate(lpg2rdfConfig, validDataType);
+        final NodeMapping nodeMapping = getNodeMapping(lpg2Rdf, lpg2rdfConfig);
+
+
+        return new LPG2RDFConfigurationImpl(label, nodeMapping);
+    }
+
+    public Node getLabelPredicate(final Resource lpg2rdfConfig, final String validDataType){
+        if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+
+        final StmtIterator labelPredicateIterator = lpg2rdfConfig.listProperties(LPG2RDF.labelPredicate);
+
+        if(!labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+        final RDFNode labelPredicateObj = labelPredicateIterator.next().getObject();
+        if(labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("More than one labelPredicate!");
+        }
+        if (!labelPredicateObj.asLiteral().getDatatypeURI().equals(validDataType)){
+            throw new IllegalArgumentException("labelPredicate is invalid!");
+        }
+        final String label = labelPredicateObj.asLiteral().getString();
+
+        return NodeFactory.createURI(label);
```

Comment:
Does this work even if the string value of the `labelPredicate` property is not actually a URI? For instance, if it is an arbitrary string such as "Hello World!" ?

I really don't know. So, I am just curious. Maybe you can check this with a unit test. If it works with arbitrary strings, that's not good and you should incorporate a check of the string first (i.e., to check that really represents a URI). To do this check you can use the Java class `URI` (in package `java.net`), which has a static function called `create`. Use this function and see whether it throws an exception.

Reply:
I have checked this. It generated no errors with strings like 'Hello World!'. I made changes based on the URI class. Now, 'Hello World!' is unacceptable, but a string like 'test' would be acceptable. So, if I consider it a URI, the current implementation seems correct. However, if I want to assume 'test' is an invalid URI, I should convert it into a URL and check it by calling `toURL().` Please let me know if the `current` implementation is correct.